app/trello_client.py

```diff
@@ -0,0 +1,111 @@
+from datetime import datetime
+import requests
+import json
+from app.to_do_item import ToDoItem
+from app.flask_config import Config
+
+
+class TrelloClient:
+    
+    def __init__(self):
+        self.config: Config = Config()
+        self.base_request_url = 'https://api.trello.com/1/boards/'+self.config.boardId+'/'
+        self.request_credentials = f'?key={self.config.key}&token={self.config.token}'
+
+    def get_items(self) -> list:
+        """
+        Fetches all saved items from the trello api.
+
+        Returns:
+            list: The list of items.
+        """
+        
+        cards_request = requests.get(self.base_request_url+'cards'+self.request_credentials)
+
+        cards_json = json.loads(cards_request.content)
+        list_id_dict = self.get_lists()
+
+        items = []
+        for node in cards_json:
+            date_in_datetime = datetime.strptime(node['dateLastActivity'], '%Y-%m-%dT%H:%M:%S.%fZ')
+            items.append(ToDoItem(node['id'], list_id_dict[node['idList']], node['name'], date_in_datetime))  
+        
+        return items
+
+    def get_item(self, id) -> dict:
+        """
+        Fetches the saved item with the specified ID.
+
+        Args:
+            id: The ID of the item.
+
+        Returns:
+            item: The saved item, or None if no items match the specified ID.
+        """
+        items = TrelloClient.get_lists()
+        return next((item for item in items if item['id'] == int(id)), None)
+
+
+    def add_item(self, title):
+        """
+        Adds a new item with the specified title to the board.
+
+        Args:
+            title: The title of the item.
+        """
+
+        json_list = self.get_lists()
```

Comment:
`json_list` is a misleading name. It's actually a dictionary (of list ids to names).

I think it would have been a bit more intuitive if `get_lists` did return a list of Trello "list" objects. The for-loop below would just have to be changed to:

```
for list in lists:
    if list['name'] == 'To Do':
        todo_list = list['id']
```